doc/dev/hmsl/hashicorp-vault.md

```diff
@@ -7,76 +7,42 @@ The `ggshield check-secret-manager hashicorp-vault` command interacts with Hashi
 Like other tests, a lot of the tests for ggshield.verticals.hmsl.secret_manager.hashicorp_vault.api_client
 are using cassettes recorded using [VCR.py](https://github.com/kevin1024/vcrpy) to replay network interactions.
 
-### Running Vault server locally with Docker
-
 To generate these cassettes, a local Hashicorp Vault instance in dev mode is used by running the official Vault docker image.
+The server is populated with test data:
 
-To run the server, first save the following configuration as `vault_config.json`:
-
-```json
-{
-  "storage": {
-    "file": {
-      "path": "/vault/file"
-    }
-  },
-  "listener": [
-    {
-      "tcp": {
-        "address": "0.0.0.0:6789",
-        "tls_disable": true
-      }
-    }
-  ],
-  "default_lease_ttl": "168h",
-  "max_lease_ttl": "720h",
-  "ui": true
-}
-```
-
-And the following script as `run_server.sh`:
-
-```bash
-#!/usr/bin/env bash
-
-set -euo pipefail
-
-echo 'Starting vault on http://127.0.0.1:6789'
-
-docker run --rm \
-    --name=testing-vault \
-    -e "VAULT_LOCAL_CONFIG=$(< vault_config.json)" \
-    -e "SKIP_SETCAP=1" \
-    -p 6789:6789 \
-    hashicorp/vault
-```
-
-You can then run the server by executing `run_server.sh`.
-
-The root token to use to interact with the web UI will be printed to stdout:
-
-```
-The unseal key and root token are displayed below in case you want to
-seal/unseal the Vault or re-authenticate.
-
-Unseal Key: <unseal_key>
-Root Token: <root_token>
-```
+- a `secret_v1` kv mount (version 1)
+- a `secret` kv mount (version 2)
+  These two mounts are populated with some file and directories (same data for both)
 
-### Using the server in tests
+The server has two tokens:
+
+- `my_vault_token` is a root token with all rights
+- `restricted_token` is a token with the `default` policy.
+  It cannot read the kv mounts (if you want to test error handling for example).
+
+These vault tokens do not use the `hvs.<random_characters>` pattern on purpose to avoid
+triggering ggshield.
+
+### Running the server
+
+The script handling the server is `scripts/hmsl/hashicorp_vault.py`:
 
-To add, remove or update secrets and mounts, you can use the web UI located at http://localhost:6789/ with the root token.
+- to start the server: `python scripts/hmsl/hashicorp_vault.py start`.
```

Comment:
If you `chmod +x` the script then it can be started without prepreding `python` to it.